src/components/table/__spec__.js

```diff
@@ -1019,20 +1021,22 @@ describe('Table', () => {
     );
 
     it('include correct component, element and role data tags', () => {
+      // TODO - 1007 - this function is checking for a 'data-component' prop which does not exist here.
+    // data-component doesnt exist as prop on mount
       rootTagTest(wrapper, 'table', 'bar', 'baz');
     });
   });
 
   describe("theme", () => {
     it("renders a --secondary if the theme is set to 'secondary'", () => {
-      const wrapper = shallow(
+      const wrapper = mount(
         <Table theme='secondary' />
       );
       expect(wrapper.find('.carbon-table--secondary').exists()).toBeTruthy();
     });
 
     it("renders a --primary if the theme is missing (default)", () => {
-      const wrapper = shallow(
+      const wrapper = mount(
```

Comment:
Rather than changing all of these to `mount`, we could add an early return to this function if `this._table` does not exist https://github.com/Sage/carbon/blob/master/src/components/table/table.js#L747